index.js

```diff
@@ -11,16 +14,23 @@ exports.handler = async (event) => {
     const query = event.queryStringParameters || {}
     const headers = util.keysToLowerCase(event.headers) || {}
     const identity = (event.requestContext || {}).identity || {}
+    const userId = cookie.parse(headers['cookie'] || '')[COOKIE_NAME]
 
     if (event.httpMethod !== 'GET' && event.httpMethod !== 'HEAD') {
       return util.text(405, 'Method not allowed')
     } else if (event.path === '' || event.path === '/') {
       return pages.home()
     } else if (event.path === '/admin') {
       return await pages.admin(query, headers)
-    } else if (event.path === '/i.gif') {
-      await tracker.log(query, headers, identity)
+    } else if ((event.path === '/i.gif' && userId) || event.path === '/r.gif') {
+      await tracker.log(query, headers, identity, userId)
       return util.pixel()
+    } else if (event.path === '/i.gif') {
+      const params = util.encodeQuery(query)
+      const domain = headers['host']
+      const expires = util.future(1)
+      const setCookie = cookie.serialize(COOKIE_NAME, uuidv4(), {domain, expires})
+      return util.redirect(`/r.gif${params}`, setCookie)
```

Comment:
I can't think of a case where a UA would hit a redirect and *not* come out on the other side.  So not logging anything here shouldn't miss useful analytics.

Reply:
If their app/browser doesn't support cookies, I still want to log their visit.  This redirect-to-check-for-cookie-support flow was an alternative to doing some sort of user-agent parsing and guessing if they supported cookies.  (But yeah, very much an edge case these days).